enumerable.rb

```diff
@@ -1,87 +1,163 @@
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      instance_of?(Array) ? yield(to_a[i]) : yield(to_a[i][0], to_a[i][1])
+      i += 1
     end
+    self
   end
 
   def my_each_with_index
-    for i in self do
-      self.class == Array ? yield(i, self.index(i)) : yield([i[0], i[1]], self.keys.index(i[0]))
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      instance_of?(Array) ? yield(to_a[i], i) : yield([to_a[i][0], to_a[i][1]], i)
+      i += 1
     end
+    self
   end
 
   def my_select
+    return to_enum(:my_select) unless block_given?
+
     new_arr = []
     new_hash = {}
-    self.my_each do |k,v|
-        if self.class == Array
-            new_arr << k if yield(k) == true
-        else
-            new_hash.store(k,v) if yield(k,v) == true
-        end
+    my_each do |k, v|
+      if instance_of?(Array)
+        new_arr << k if yield(k) == true
+      elsif yield(k, v) == true
+        new_hash.store(k, v)
+      end
     end
     new_arr.empty? ? new_hash : new_arr
   end
 
   def my_all?(*args)
     if block_given?
-      self.my_each do |e|
+      my_each do |e|
         return false if yield(e) == false
       end
     end
-    if !block_given? 
-      if self.index(nil)
-        return !self.index(nil) ? true : false
-      end  
-      if !args.empty?
-        if ( args[0].class != Regexp) && ( args[0].class != Class)
-          return  false 
-        end
-        self.my_each do |e|
-          if ( args[0].class == Regexp) &&(args[0].match(e.to_s)) == nil
-            return false
-          end
-          if ( args[0].class == Class) && (e.is_a? args[0]) == false 
-            return false
-          end
+    unless block_given?
+      if index(nil)
+        return !index(nil) ? true : false
+      end
+
+      unless args.empty?
+        return false if (args[0].class != Regexp) && (args[0].class != Class)
+
+        my_each do |e|
+          return false if args[0].instance_of?(Regexp) && args[0].match(e.to_s).nil?
+          return false if args[0].instance_of?(Class) && (e.is_a? args[0]) == false
         end
       end
     end
-    return true
+    true
   end
-end
-
 
-# p [1, 5, 3.14].my_all?(5)
-# p [1, 5, 3.14].all?(5)
+  def my_any?(*args)
+    if block_given? == false
+      if args.empty?
+        !empty? ? my_each { |v| return true unless v.nil? } == false : false
+      else
+        return my_each { |v| return true if v.is_a?(*args) } == false if args[0].instance_of?(Class)
+        return my_each { |v| return true if v.match(args[0]) } == false if args[0].instance_of?(Regexp)
 
-# p [nil, nil, nil].my_all?
-# p [nil, nil, nil].all?
-
-# p ["some", "somithimes", "something"].my_all?(/s/)
-# p ["some", "somithimes", "something"].all?(/s/)
+        my_each { |v| return true if v == args[0] } == false
+      end
+    else
+      my_each { |v| return true if yield(v) } == false
+    end
+  end
```

Comment:
- [ ] returns false if the block ever returns a value false or nil::range
```
  block = proc { |num| num > 9 }
  range = Range.new(5, 50)
  p range.my_any?(&block) # false
  p range.any?(&block) # true
```

- [ ]  when no block or argument is given returns false if at least one of the collection is not true
```
  p [nil, false, nil, false].any? # false
  p [nil, false, nil, false].my_any? # true
```